models/clip/clip_src/clip.py

```diff
@@ -0,0 +1,233 @@
+import hashlib
```

Comment:
TODO for myself: add reference to where this code was taken from.

@Kamino666 is it precisely https://github.com/openai/CLIP/blob/main/clip/clip.py without a change?

Reply:
No, I change some codes.

Reply:
Ok, let me merge it to the `main` and we need to add this reference + a sentence that describes the changes on a high level